src/lib/async-handlers.js

```diff
@@ -0,0 +1,25 @@
+exports.asyncHandler = fn => {
+	if (fn.length === 4) {
+		return (err, req, res, next) => {
+			const result = fn(err, req, res, next);
+			if (!result.headersSent && result instanceof Promise) {
+				result.catch(next);
+			}
+		};
+	} else if (fn.length < 4) {
+		return (req, res, next) => {
+			const result = fn(req, res, next);
+			if (!result.headersSent && result instanceof Promise) {
```

Comment:
Perhaps this should explicitly throw a type error if the handler does not return either the response object or a promise?

Reply:
Maybe. 1 idea I have had is to limit this to only support `async` functions and reject in `fn` isnt an `AsyncFunction`.

This would mean it wouldn't work for functions returning promises in general though.